GitCommands/Git/GitModule.cs

```diff
@@ -1035,13 +1041,52 @@ public Process RunBash(string bashCommand = null)
                 {
                     args = "--login -i -c \"" + bashCommand.Replace("\"", "\\\"") + "\"";
                 }
+                args = "/c \"\"{0}\" " + args;
 
-                string termCmd = File.Exists(AppSettings.GitBinDir + "bash.exe") ? "bash" : "sh";
-                return RunExternalCmdDetachedShowConsole("cmd.exe",
-                    string.Format("/c \"\"{0}{1}\" {2}", AppSettings.GitBinDir, termCmd, args));
+                if (TryFindShellPath("bash.exe", out shellPath))
+                {
+                    return RunExternalCmdDetachedShowConsole("cmd.exe", string.Format(args, shellPath));
+                }
+
+                if (TryFindShellPath("sh.exe", out shellPath))
+                {
+                    return RunExternalCmdDetachedShowConsole("cmd.exe", string.Format(args, shellPath));
+                }
+
+                return RunExternalCmdDetachedShowConsole("cmd.exe", @"/K echo git bash command not found! :(");
             }
         }
 
+        private bool TryFindShellPath(string shell, out string shellPath)
+        {
+            shellPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "Git", shell);
+            if (File.Exists(shellPath))
+                return true;
+
+            shellPath = AppSettings.GitBinDir + shell;
+            if (File.Exists(shellPath))
+                return true;
+
+            shellPath = GetShellInPathEnvironmentVariable(shell);
+            if (shellPath != null)
+                return true;
+
+            shellPath = null;
+            return false;
+        }
+
+        public static string GetShellInPathEnvironmentVariable(string fileName)
```

Comment:
According to #2924, can you use [TryFindFullPath](https://github.com/gitextensions/gitextensions/commit/2bde919d88b65b98911ab04b0707e24141533291#diff-08f0370985404e7026a5e6a2f45e665eR186)?